Updates for strict account linking (#5107)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md

    content: Okta supports authentication with Login.gov as an external Identity Provider. Set up the Login.gov IdP using OpenID Connect with private key JWT in Okta.
---

> This document explains how to configure Login.gov as an [external Identity Provider](/docs/concepts/identity-providers/) for your app. You do this by creating an Identity Provider in Okta, creating an app at Login.gov, testing the configuration, and creating a Login.gov sign-in option.

Reviewer: ```suggestion
This document explains how to configure Login.gov as an [external Identity Provider](/docs/concepts/identity-providers/) for your app. You do this by creating an IdP in Okta, creating an app at Login.gov, testing the configuration, and creating a Login.gov sign-in option.
```

---

#### Learning outcomes

> Configure Login.gov as an external Identity Provider so that your users can quickly sign up or sign in to your app using their Login.gov account.

Reviewer: ```suggestion
Configure Login.gov as an external IdP so that your users can quickly sign up or sign in to your app using their Login.gov account.
```

> **Note:** Okta also supports other services, such as directories and credential providers. See the [Okta Integration Network Catalog](https://www.okta.com/okta-integration-network/) to browse all integrations by use case.

## Create an Identity Provider in Okta

Login.gov requires you to [test your app integration](https://developers.login.gov/testing/) in their identity sandbox before you can enable it in production. As a result, you have to first create a **Login.gov IdP (Sandbox)** configuration in Okta to test your app integration. When you're ready for [Login.gov production deployment](https://developers.login.gov/production/), you can then create the production **Login.gov IdP** configuration in Okta.

> > **Note:** See the [Identity Providers API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/IdentityProvider/#tag/IdentityProvider/operation/createIdentityProvider) for request and response examples of creating an Identity Provider in Okta using the API.

Reviewer: ```suggestion
> **Note:** See the [Identity Providers API](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/IdentityProvider/#tag/IdentityProvider/operation/createIdentityProvider) for request and response examples of creating an IdP in Okta using the API.
```

   * **Scopes**: Leave the defaults (`profile`, `profile:name`, `email`) for IAL1 assurance. These scopes are included when Okta makes an OpenID Connect request to the IdP. See [Login.gov OIDC scopes for required attributes](https://developers.login.gov/attributes/).

   * **Type of Identity Verification**: The [type of service level](https://developers.login.gov/oidc/authorization/#request-parameters) available for this app. Select **ial/1** for standard MFA-protected email-based sign-in flows.
   * **AAL value**: Select the [authentication assurance level](https://developers.login.gov/oidc/authorization/#request-parameters) required.
      > **Note:** If you leave the default value, the IdP requires a user to be authenticated with a second factor: `urn:gov:gsa:ac:classes:sp:PasswordProtectedTransport:duo`.

   In the optional **Authentication Settings** section:

>    * **IdP username**: This is the expression (written in Okta Expression Language) that is used to convert an IdP attribute to the app user's `username`. This IdP username is used for matching an app user to an Okta User.

Reviewer: ```suggestion
   * **IdP username**: This is the expression (written in Okta Expression Language) that is used to convert an IdP attribute to the app user's `username`. This IdP username is used for matching an app user to an Okta user.
```

If something is configured incorrectly, the authorization response contains error information to help you resolve the issue.

> The Okta Identity Provider that you created generates an authorize URL with several blank parameters that you can fill in to test the flow with the Identity Provider. For example:

Reviewer: ```suggestion
The Okta IdP that you created generates an authorize URL with several blank parameters that you can fill in to test the flow with the IdP. For example:
```

  nonce={nonce}
```

> **Note:** If you’re using Authorization Code with PKCE as the grant type for your client app, you must generate and store the PKCE. See [Implement authorization by grant type](/docs/guides/implement-grant-type/authcodepkce/main/#flow-specifics). Okta recommends that you use the [AuthJS SDK](https://github.com/okta/okta-auth-js#signinwithredirectoptions) with this grant type.

In the URL, replace `{yourOktaDomain}` with your org's base URL, and then replace the following values:

* `idp`: Your `{yourIdPId}` value that you obtained from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).

> * `client_id`: Use the `{clientId}` value that you obtained from your OpenID Connect client app. This isn't the **Client ID** value from the Identity Provider.

Reviewer: ```suggestion
* `client_id`: Use the `{clientId}` value that you obtained from your OpenID Connect client app. This isn't the **Client ID** value from the IdP.
```

```

After the user clicks the link, they’re prompted to sign in with Login.gov. After the user successfully signs in, the user is returned to the specified `redirect_uri` along with an ID token in JWT format.

### Use the AuthJS SDK

If you don't want pre-built views, or need deeper levels of customization, then you can use the same AuthJS SDK that the Sign-In Widget is built with. See the `idp` option in the [AuthJS SDK GitHub repo](https://github.com/okta/okta-auth-js#usage-guide).

## Next steps

* To map Okta attributes to app attributes, use the [Profile Editor](https://help.okta.com/okta_help.htm?id=ext_app_map).

> * To add another Identity Provider, start by choosing an [external Identity Provider](/docs/guides/identity-providers/).

Reviewer: ```suggestion
* To add another IdP, start by choosing an [external Identity Provider](/docs/guides/identity-providers/).
```
